src/wagtail_ai/static_src/AIControl.tsx

```diff
@@ -195,15 +161,34 @@ function AIControl({
     setIsLoading(false);
   };
 
+  const handleClick = (e: any): void => {
+    if (!e.target) {
+      return;
+    }
+    if (!draftailEditorWrapperRef?.current) {
+      // Reference the Draftail editor wrapper so we can insert error messages and loading overlays
+      draftailEditorWrapperRef.current = e.target.closest(
```

Comment:
**Issue:**
Referencing the `data-draftail-editor-wrapper` is difficult, using the `registerPlugin` method as a replacement for the removed `initEditorProxy` code. 


**Solutions**

1. The approach here (commited code), extend Draftail's ToolbarButton component to allow the onClick handler access to the event object were I can access the DOM to find the closest draftail editor, similar to how it's [currently being done in Wagtail for other components](https://github.com/search?q=repo%3Awagtail%2Fwagtail%20data-draftail-editor-wrapper&type=code).


2. Wrapping this component in a div instead of a template fragment to use useRef and the `closest` method on. at the moment I can't call the function `.closest(` on a react element such as the ToolbarButton, but it seems i can on a standard div.
```
aIControlRef.current.closest('[data-draftail-editor-wrapper]')
...
 return (
    <div ref={aIControlRef}>
      <ToolbarButton
```
3. Refactor to use an [arbitrary-plugin](https://docs.wagtail.org/en/v5.2.1/extending/extending_draftail.html#arbitrary-plugins), I think this would complicate the code more, and I'm not sure if this is the answer, but it looks like I might be able to access [getEditorRef](https://github.com/draft-js-plugins/draft-js-plugins/blob/master/HOW_TO_CREATE_A_PLUGIN.md) which might let me reference the editor wrapper in a more standard way.

4. Similar to option 3, perhaps theres a hook or method provided by Draftail that I've missed?

**Solutions involving fixes upstream**

1. With guidance opening an issue in Draftail to support passing extra props for [ControlComponentProps](https://github.com/springload/draftail/blob/33c9bfad7d63e5ae1367eb9d5178a0424cba8c0b/src/api/types.ts#L137) to allow referencing the editor as the current props seem limiting.

2. Update wagtail's InitEditor code to pass in reference to `data-draftail-editor-wrap` for all "PLUGINS", as it looks to be wagtail specific and is [used in a few places this way](https://github.com/search?q=repo%3Awagtail%2Fwagtail%20data-draftail-editor-wrapper&type=code), so might also be an opportunity to improve the codebase.

Reply:
From working on styling improvements related to this component, I'd like to go with option 2, this would give me more control over the dropdown component (adding position: relative).